int32 stop_flag
int32 bbox_size